harness/variables.go

```diff
@@ -0,0 +1,102 @@
+package harness
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+const (
+	// ErrorMessageKey is the key used to retrieve or store the error message content.
+	ErrorMessageKey = "ERROR_MESSAGE"
+
+	// ErrorCodeKey is the key used to identify the specific error code associated with an error.
+	ErrorCodeKey = "ERROR_CODE"
+
+	// ErrorCategoryKey is the key used to classify the category of the error, which can help in grouping similar types of errors.
+	ErrorCategoryKey = "ERROR_CATEGORY"
+
+	// MetadataFile is the key for the file that stores metadata associated with an error, such as details about the error's source or context.
+	MetadataFile = "ERROR_METADATA_FILE"
+
+	// DroneOutputFile is the key for the file where output can exported and utilized in the subsequent steps in Harness CI pipeline.
+	DroneOutputFile = "DRONE_OUTPUT"
+)
+
+// SetSecret sets a new secret by adding it to the DRONE_OUTPUT file
+func SetSecret(name, value string) error {
+	return WriteEnvToOutputFile(DroneOutputFile, name, value)
+}
+
+// UpdateSecret updates an existing secret with a new value in the DRONE_OUTPUT file
+func UpdateSecret(name, value string) error {
+	return WriteEnvToOutputFile(DroneOutputFile, name, value)
+}
+
+// DeleteSecret removes a secret by setting it to an empty value in the DRONE_OUTPUT file
+func DeleteSecret(name string) error {
+	return WriteEnvToOutputFile(DroneOutputFile, name, "")
+}
+
+// SetError sets the error message and error code, writing them to the CI_ERROR_METADATA file
+// SetError sets the error message, error code, and error category, writing them to the CI_ERROR_METADATA file
+func SetError(message, code, category string) error {
```

Comment:
Nit: we can add overloaded method with just message as parameter and default error code and default error category as they are not yet supported . this method can be used in the plugins till then